services/app-api/src/emailer/emails/sendRateQuestionStateEmail.ts

```diff
@@ -0,0 +1,99 @@
+import type { RateQuestionType, RateType } from '../../domain-models'
+import type { EmailConfiguration, EmailData } from '../emailer'
+import {
+    getActuaryContactEmails,
+    renderTemplate,
+    stripHTMLFromTemplate,
+} from '../templateHelpers'
+import { formatCalendarDate } from '../../common-code/dateHelpers'
+import { pruneDuplicateEmails } from '../formatters'
+import { rateQuestionResponseURL } from '../generateURLs'
+
+export const sendRateQuestionStateEmail = async (
+    rate: RateType,
+    config: EmailConfiguration,
+    rateQuestion: RateQuestionType
+): Promise<EmailData | Error> => {
+    const contractRevisions = rate.packageSubmissions[0].contractRevisions
+    const rateFormData = rate.packageSubmissions[0].rateRevision.formData
+    const parentContractID = rate.parentContractID
+    const shouldIncludeActuaries =
+        rateFormData.actuaryCommunicationPreference === 'OACT_TO_ACTUARY'
+
+    if (!rateFormData.rateCertificationName) {
+        return new Error(
+            'Error getting rate name. Rate certification name was undefined.'
+        )
+    }
+
+    const submitterEmails = contractRevisions.reduce(
+        (contacts: string[], cr) => {
+            if (cr.submitInfo?.updatedBy.email) {
+                return contacts.concat(cr.submitInfo.updatedBy.email)
+            }
+            return contacts
+        },
+        []
+    )
+
+    const stateContactEmails = contractRevisions.reduce(
+        (contacts: string[], cr) => {
+            const stateContacts: string[] = []
+            if (cr.formData.stateContacts.length) {
+                cr.formData.stateContacts.forEach((stateContact) => {
+                    if (stateContact.email) {
+                        stateContacts.push(stateContact.email)
+                    }
+                })
+            }
+            return contacts.concat(stateContacts)
+        },
+        []
+    )
+
+    const actuaryEmails = shouldIncludeActuaries
+        ? getActuaryContactEmails(rateFormData)
+        : []
+
+    const toAddresses = pruneDuplicateEmails([
```

Comment:
Random though - Seeing this with fresh eyes today, haven't looked at emailer for awhile.

I am seeing copypasta code and thinking how more of it could be abstracted up to the emailer level  rather than handling in each emailer function. Some of this code maybe goes instead into `sendSESEmail`. 

Things like 
- it should never be possible to send to same address twice
- the subject line logic has pre-pended the environment

Reply:
Maybe this could be a RSP topic, but I rather like keeping this kind of logic contained inside the functions themselves.